task2/src/components/home/MovieCard/MovieCard.jsx

```diff
@@ -1,29 +1,84 @@
+import {
+    useState,
+    useRef
+} from 'react';
 import PropTypes from 'prop-types';
-import { 
+import {
+    EditMovieModal,
+    DeleteMovieModal
+} from '@components/home';
+import {
     MovieCard as Wrapper,
     MovieCardImage,
     MovieCardInfo,
     MovieCardTitle,
     MovieCardReleaseDate,
-    MovieCardGenres
+    MovieCardGenres,
+    MovieCardDropdownButton,
+    MovieCardDropdownOption
 } from './MovieCard.styles';
 
-const MovieCard = (props) => (
-    <Wrapper>
-        <MovieCardImage src={props.movie.poster}/>
-        <MovieCardInfo>
-            <MovieCardTitle>
-                {props.movie.title}
-            </MovieCardTitle>
-            <MovieCardReleaseDate>
-                {props.movie.releaseDate.getFullYear()}
-            </MovieCardReleaseDate>
-            <MovieCardGenres>
-                {props.movie.genres}
-            </MovieCardGenres>
-        </MovieCardInfo>
-    </Wrapper>
-);
+const MovieCard = (props) => {
+    const [movie, setMovie] = useState(props.movie);
+
+    const editMovieModalRef = useRef(null);
+    const closeEditButtonRef = useRef(null);
```

Comment:
I'm not sure that we really need to use `ref`s everywhere. Do we really need it?

Reply:
the same question)

Reply:
I thought of using it as custom hooks. But Dima said that it is another thing. Left it for further actions.